hls4ml/converters/onnx_to_hls.py

```diff
@@ -125,252 +183,95 @@ def onnx_to_hls(yamlConfig):
-                layer['pad_top'] = pads[0]
-                layer['pad_bottom'] = pads[2]
-                layer['pad_left'] = pads[1]
-                layer['pad_right'] = pads[3]
-
-                if all(x == 0 for x in pads): # No padding, i.e., 'VALID' padding in Keras/Tensorflow
-                    layer['out_width'] = int(math.ceil(float(layer['in_width'] - layer['filt_width'] + 1) / float(layer['stride_width'])))
-                    layer['out_height'] = int(math.ceil(float(layer['in_height'] - layer['filt_height'] + 1) / float(layer['stride_height'])))
-                else:
-                    layer['out_height'] = int(math.ceil(float(layer['in_height']) / float(layer['stride_height'])))
-                    layer['out_width'] = int(math.ceil(float(layer['in_width']) / float(layer['stride_width'])))
-
-                layer['n_out'] = layer['out_height'] * layer['out_height'] * layer['n_filt']
-                current_shape=[current_shape[0], layer['n_filt'], layer['out_height'], layer['out_width']]
-        elif layer['class_name'] in ['ELU', 'LeakyReLU', 'ThresholdedReLU']:
-            layer['activation'] = layer['class_name']
-            layer['activ_param'] = get_onnx_attribute(operation, 'alpha', 0.01)
-        elif layer['class_name']=='PReLU':
-            layer['activation'] = layer['class_name']
-
-        elif layer['class_name'] in [operation_map.get(op, op) for op in merge_operations]:
-            layer['op'] = layer['class_name'].lower()
-            if layer['class_name'] == 'Concatenate':
-                rank = len(current_shape[1:])
-                if rank > 3:
-                    raise Exception('ERROR: Concatenation of tensors with rank > 3 is not yet supported.')
-                layer['op'] = layer['class_name'].lower() + '{}d'.format(rank)
-                layer['axis'] = get_onnx_attribute(operation, 'axis')
-            else:
-                layer['class_name'] = 'Merge'
-            layer['inputs'] = [inputs_map.get(x, x) for x in operation.input]
-            if len(layer['inputs']) > 2:
-                raise Exception('ERROR: Merging more than two tensors is not yet supported.')
-
+                output_shape = current_shape[0]
+            continue 
+        
+        if node.op_type in supported_layers:
+            layer_counter = layer_counter + 1
+        
+        #Process the layer
+        layer, output_shape = layer_handlers[node.op_type](reader, node, inputs_map, current_shape, graph, config)
+        
         sanitize_layer_name(layer)
         print('Layer name: {}, layer type: {}, current shape: {}'.format(layer['name'], layer['class_name'], current_shape))
-        layer_list.append( layer )
+        layer_list.append(layer)
 
 
     #################
     ## Generate HLS
     #################
 
     print('Creating HLS model')
-    hls_model = HLSModel(yamlConfig, reader, layer_list, input_layers, output_layers)
+    hls_model = HLSModel(config, reader, layer_list)
```

Comment:
If you don't pass input/output layers only sequential models are supported. This should be solved now or documented as a limitation in the docs for future improvement